lib/web_ui/dev/steps/run_tests_step.dart

```diff
@@ -164,6 +175,54 @@ class RunTestsStep implements PipelineStep {
     }
     return message.toString();
   }
+
+  Future<SkiaGoldClient?> _createSkiaClient() async {
+    const FileSystem fs = LocalFileSystem();
+    final Directory goldensRoot =
+        fs.directory(environment.webUiSkiaGoldDirectory.path);
+    final SkiaGoldClient skiaClient =
+        SkiaGoldClient(goldensRoot, browserName: browserName);
+
+    if (!await _checkSkiaClient(skiaClient)) {
+      print('WARNING: Unable to use Skia Client in this environment.');
+      return null;
+    }
+
+    return skiaClient;
+  }
+
+  /// Checks whether the Skia Client is usable in this environment.
+  Future<bool> _checkSkiaClient(SkiaGoldClient skiaClient) async {
+    const FileSystem fs = LocalFileSystem();
+    final Directory goldensRoot =
+        fs.directory(environment.webUiSkiaGoldDirectory.path);
+    final SkiaGoldClient skiaClient =
+        SkiaGoldClient(goldensRoot, browserName: browserName);
+
+    // Now let's check whether Skia Gold is reachable or not.
+    if (isLuci) {
+      if (SkiaGoldClient.isAvailable) {
+        try {
+          await skiaClient.auth();
+          return true;
+        } catch (e) {
+          print(e);
+        }
+      }
+    } else {
+      try {
+        // Check if we can reach Gold.
+        await skiaClient.getExpectationForTest('');
+        return true;
+      } on io.OSError catch (_) {
+        print('OSError occurred, could not reach Gold.');
+      } on io.SocketException catch (_) {
+        print('SocketException occurred, could not reach Gold.');
```

Comment:
These should do `io.stderr.writeln`, but as mentioned above, I think I'd prefer a hard failure in this case. Instead, if whether goldens are compared should be controlled by a command-line flag.

Reply:
I don't think this line of code is the right place to do it, but I agree that we should fail if Skia Gold isn't available and we expect golden tests to run.

Reply:
Right now, I don't want to introduce any failures caused by Skia Gold. In a future step, I'm going to start relying on Skia Gold to block PRs (until images are triaged). That's when I'm going to add the check to make sure Skia Gold is actually running when we expect it to.

Reply:
SGTM. Let's file an issue and leave a TODO pointing to it.

Reply:
Was there an issue filed & TODO added?

Reply:
I updated the [issue](https://github.com/flutter/flutter/issues/62706) to list all steps required to finish this work.